frb_codegen/src/parser/mod.rs

```diff
@@ -343,6 +377,53 @@ fn item_method_to_function(
     if let Type::Path(p) = item_impl.self_ty.as_ref() {
         let struct_name = p.path.segments.first().unwrap().ident.to_string();
         let span = item_method.sig.ident.span();
+
+        // get/check inputs for mutability first
+        let mut is_mut = false;
+        // let inputs: syn::punctuated::Punctuated<_, _> = item_method
+        let inputs: Punctuated<FnArg, Comma> = item_method
+            .sig
+            .inputs
+            .iter()
+            .enumerate()
+            .map(|(i, input)| -> ParserResult<_> {
+                let mut segments = Punctuated::new();
+                segments.push(PathSegment {
+                    ident: Ident::new(struct_name.as_str(), span),
+                    arguments: PathArguments::None,
+                });
+                if let FnArg::Receiver(Receiver { mutability, .. }) = input {
+                    if i == 0 && mutability.is_some() {
+                        is_mut = true;
```

Comment:
nit:

1. do we really need these logic about mutability (i.e. "skip if mut") - if someone wrongly make it mut, I guess our bridge_generated.rs will just fail to compile (since we will generate code like "call a mutable method from non-mutable object")?
2. If yes, consider the code `fn f(mut self) {}`. it seems that it will be marked as mutable and thus skip generating, but indeed it is totally acceptable (since it is `mut self` instead of `&mut self`? (brainstorm, correct me if I am wrong!)

Reply:
I make the logic to skip generating methods with `&mut self`, since it is not clear for frb to generate corresponding code for it or `mut self`(correct me if I am wrong).

And the reason for keeping the logic is that a struct in rust may be extensively used in rust side, while it would still need to be exposed to dart side. In this case, I think it is unreasonable to force users not define methods with `&mut self` or `mut self`.

In other word, once frb could genearte methods with `mut`, this logic could be erased.

Reply:
IMHO, `mut self` means (1) just a `self`, i.e. take the ownership of the object (2) but at the same time, allow to mutate it.

It is equivalent to:

```rust
fn f(mut self) { ... }

// equivalent to:
fn f(self) {
  let mut self2 = self;
  ...
}
```

so frb will not be unhappy imho

I made a demo:

```rust
struct A {
    x: i32,
}

impl A {
    fn f(mut self) {
        self.x = 2;
        println!("self.x is {}", self.x);
    }
}

fn main() {
    let a = A {x: 1};
    a.f();
}
```

yields 

```
self.x is 2
```

Reply:
Anyway, this is not important (not the main point of this PR), so just ignore it.

Reply:
For the demo, it is syntactically correct. But then the `a` can't be used any longer, because there is no ownership:
```rust
fn main() {
    let a = A { x: 1 };
    a.f();
    a.f();// not available for any method, because there is no ownership
}
```
Besides, currently methods of `&self` are feasible, while `mut self` and `&mut self` are not, so I skipped them with `log::warn`.